pandas-stubs/core/algorithms.pyi

```diff
@@ -8,24 +8,21 @@ import pandas as pd
 from pandas import (
     Categorical,
     CategoricalIndex,
-    DatetimeIndex,
     Index,
+    IntervalIndex,
     PeriodIndex,
-    RangeIndex,
     Series,
 )
 from pandas.api.extensions import ExtensionArray
 
 from pandas._typing import AnyArrayLike
 
-@overload
-def unique(values: DatetimeIndex) -> DatetimeIndex: ...
 @overload
 def unique(values: PeriodIndex) -> PeriodIndex: ...
 @overload
 def unique(values: CategoricalIndex) -> CategoricalIndex: ...
 @overload
-def unique(values: RangeIndex | pd.Float64Index) -> np.ndarray: ...
+def unique(values: IntervalIndex) -> IntervalIndex: ...
 @overload
 def unique(values: Index) -> Index | np.ndarray: ...
```

Comment:
Since you have the cases where we return specific index types called out, could we make this return `np.ndarray`?  If mypy complains about different return types, I'd be OK with putting a `#type ignore` on the right overload.

Reply:
Fixed this with ignores.